teacher.rb

```diff
@@ -0,0 +1,16 @@
+require_relative 'person'
+
+# The Teacher class represents a teacher with a name, age, and specialization.
+# It inherits from the Person class and overrides the method can_use_services?.
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission)
```

Comment:
- [ ] Please make sure to pass the parameters in the super method in the correct order as the parent class initializer expects.
  - You can consider the below code.
  
```ruby
super(age, name, parent_permission)